bin/reth/src/node/events.rs

```diff
@@ -79,6 +79,7 @@ impl NodeState {
                 result: ExecOutput { checkpoint, done },
             } => {
                 self.current_checkpoint = checkpoint;
+                self.latest_block = Some(checkpoint.block_number);
```

Comment:
hmm, I skeptical that this is the correct fix, because we only want to do this for the finish stage but this updates the for every stage, right?

Reply:
hmm, you're right, added a check for `Finish` stage